### PR TITLE
design settings ui, integrate settings page api

### PR DESCRIPTION
**Pull request by Snehal018**

### CODE REVIEW

**Review by bharathkumarchowdary — CHANGES_REQUESTED** [review]
- [ ] In reset password page, it is better to fetch the user's email ID from backend and keep it in textbox instead of leaving it blank

image

- [ ] Open mail app button not working

File: schema/settings.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { z } from 'zod'
+
+const editProfileSchema = z.object({
+  first_name: z.string().min(3, 'Please enter at least 3 characters!').max(150, 'Please enter at most 150 characters!'),
+  last_name: z.string().min(3, 'Please enter at least 3 characters!').max(150, 'Please enter at most 150 characters!'),
+  username: z.string().min(3, 'Please enter at least 3 characters!').max(150, 'Please enter at most 150 characters!'),
```
Comment (mohitkhatri06) [inline schema/settings.ts:6]: in the frontend this field we had minimum 6 chars

**Review by mohitkhatri06 — APPROVED** [review]

**Review by bharathkumarchowdary — CHANGES_REQUESTED** [review]
email verification flow is following reset password flow. why? also why is it not loading the user's email? why does the user have to re-enter email id?

**Comment by bharathkumarchowdary** [comment]
@Snehal018 , Please mask reset password for now, we'll re-enable it in future

**Comment by Snehal018** [comment]
@bharathkumarchowdary , do i only need to disable reset password or both reset and forgot password ?

**Comment by bharathkumarchowdary** [comment]
> @bharathkumarchowdary , do i only need to disable reset password or both reset and forgot password ?

both

**Comment by bharathkumarchowdary** [comment]
@Snehal018 , email verification is still not loading user's email and keeping the field blank

**Comment by bharathkumarchowdary** [comment]
> email verification flow is following reset password flow. why? also why is it not loading the user's email? why does the user have to re-enter email id?

**Review by mohitkhatri06 — APPROVED** [review]

**Comment by bharathkumarchowdary** [comment]
I'm getting the following error 

> ⚠️  Something went wrong running `pod install` in the `ios` directory.
Command `pod install` failed.
└─ Cause: The following Swift pods cannot yet be integrated as static libraries:
> The Swift pod `FirebaseCoreInternal` depends upon `GoogleUtilities`, which does not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.
> The Swift pod `FirebaseSessions` depends upon `FirebaseCore`, `FirebaseCoreExtension`, `FirebaseInstallations`, `GoogleDataTransport`, `GoogleUtilities`, and `nanopb`, which do not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.

**Comment by Snehal018** [comment]
@bharathkumarchowdary , why are you installing pods ? since we are using expo we should not install pods as its native thing to do not the expo way

**Comment by bharathkumarchowdary** [comment]
I'm not able to login
image

**Comment by mohitkhatri06** [comment]
> I'm not able to login image

@bharathkumarchowdary did you checked that in the poject it is ".env" instead of ".env.example"

**Comment by bharathkumarchowdary** [comment]
@mohitkhatri06 , yes it is so!

**Review by bharathkumarchowdary — CHANGES_REQUESTED** [review]
1. Even when there is no goal in user's account, it shows some on track, off track cards 
image

2. Why is KPI hardcoded single choice?
3. I'm not able to add income expense item from income statement. without that, we can't test goals
4. Completion date might be confusing. Change the name to Target Date or Target to Accomplish or Accomplish goal by or Target Completion Date. Is this the same as target_date in API?

**Comment by bharathkumarchowdary** [comment]
In PR 45, the latest PR, there is no error, but in this PR, I'm consistently seeing that it shows this error. I don't know why and I can't debug

> /opt/homebrew/bin/npm run ios
lekkalu-app@1.0.0 ios
 expo start --ios
(node:63941) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Starting project at /Users/bharathkumargundala/code/lekkalu-app
env: load .env
env: export EXPO_PUBLIC_TOKEN_BASE_URL EXPO_PUBLIC_USER_BASE_URL EXPO_PUBLIC_API_BASE_URL EXPO_PUBLIC_API_V2_BASE_URL
Experimental path aliases feature is enabled. Learn more: https://docs.expo.dev/guides/typescript/#path-aliases
Starting Metro Bundler
CommandError: No development build (com.lekkalu.finuncle) for this project is installed. Please make and install a development build on the device first.
Learn more: https://docs.expo.dev/development/build/

**Comment by Snehal018** [comment]
> 1. Even when there is no goal in user's account, it shows some on track, off track cards
>    image
> 2. Why is KPI hardcoded single choice?
> 3. I'm not able to add income expense item from income statement. without that, we can't test goals
> 4. Completion date might be confusing. Change the name to Target Date or Target to Accomplish or Accomplish goal by or Target Completion Date. Is this the same as target_date in API?

@bharathkumarchowdary, why is this PR used to test the goal feature when there is a dedicated goals PR where the feature is completely integrated? The only required change here is the name change of completion date

**Comment by bharathkumarchowdary** [comment]
> > 1. Even when there is no goal in user's account, it shows some on track, off track cards
> >    image
> > 2. Why is KPI hardcoded single choice?
> > 3. I'm not able to add income expense item from income statement. without that, we can't test goals
> > 4. Completion date might be confusing. Change the name to Target Date or Target to Accomplish or Accomplish goal by or Target Completion Date. Is this the same as target_date in API?
> 
> @bharathkumarchowdary, why is this PR used to test the goal feature when there is a dedicated goals PR where the feature is completely integrated? The only required change here is the name change of completion date

@Snehal018 , Yes it's a mistake, please don't consider [this](https://github.com/GVR-Secure-Solutions/lekkalu-app/pull/41#pullrequestreview-1873980302) comment for this PR. [This](https://github.com/GVR-Secure-Solutions/lekkalu-app/pull/41#issuecomment-1937099766) is the relevant comment

**Comment by Snehal018** [comment]
> In PR 45, the latest PR, there is no error, but in this PR, I'm consistently seeing that it shows this error. I don't know why and I can't debug
> 
> > /opt/homebrew/bin/npm run ios
> > lekkalu-app@1.0.0 ios
> > expo start --ios
> > (node:63941) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
> > (Use `node --trace-deprecation ...` to show where the warning was created)
> > Starting project at /Users/bharathkumargundala/code/lekkalu-app
> > env: load .env
> > env: export EXPO_PUBLIC_TOKEN_BASE_URL EXPO_PUBLIC_USER_BASE_URL EXPO_PUBLIC_API_BASE_URL EXPO_PUBLIC_API_V2_BASE_URL
> > Experimental path aliases feature is enabled. Learn more: https://docs.expo.dev/guides/typescript/#path-aliases
> > Starting Metro Bundler
> > CommandError: No development build (com.lekkalu.finuncle) for this project is installed. Please make and install a development build on the device first.
> > Learn more: https://docs.expo.dev/development/build/

this is an issue with the development build not being installed in the simulator

**Comment by bharathkumarchowdary** [comment]
> > In PR 45, the latest PR, there is no error, but in this PR, I'm consistently seeing that it shows this error. I don't know why and I can't debug
> > > /opt/homebrew/bin/npm run ios
> > > lekkalu-app@1.0.0 ios
> > > expo start --ios
> > > (node:63941) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
> > > (Use `node --trace-deprecation ...` to show where the warning was created)
> > > Starting project at /Users/bharathkumargundala/code/lekkalu-app
> > > env: load .env
> > > env: export EXPO_PUBLIC_TOKEN_BASE_URL EXPO_PUBLIC_USER_BASE_URL EXPO_PUBLIC_API_BASE_URL EXPO_PUBLIC_API_V2_BASE_URL
> > > Experimental path aliases feature is enabled. Learn more: https://docs.expo.dev/guides/typescript/#path-aliases
> > > Starting Metro Bundler
> > > CommandError: No development build (com.lekkalu.finuncle) for this project is installed. Please make and install a development build on the device first.
> > > Learn more: https://docs.expo.dev/development/build/
> 
> this is an issue with the development build not being installed in the simulator

@Snehal018 , If that is so, why is it working in one PR and not in another? Don't you think the problem might be in the way we mention package dependencies in git commit?

**Comment by Snehal018** [comment]
> > > In PR 45, the latest PR, there is no error, but in this PR, I'm consistently seeing that it shows this error. I don't know why and I can't debug
> > > > /opt/homebrew/bin/npm run ios
> > > > lekkalu-app@1.0.0 ios
> > > > expo start --ios
> > > > (node:63941) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
> > > > (Use `node --trace-deprecation ...` to show where the warning was created)
> > > > Starting project at /Users/bharathkumargundala/code/lekkalu-app
> > > > env: load .env
> > > > env: export EXPO_PUBLIC_TOKEN_BASE_URL EXPO_PUBLIC_USER_BASE_URL EXPO_PUBLIC_API_BASE_URL EXPO_PUBLIC_API_V2_BASE_URL
> > > > Experimental path aliases feature is enabled. Learn more: https://docs.expo.dev/guides/typescript/#path-aliases
> > > > Starting Metro Bundler
> > > > CommandError: No development build (com.lekkalu.finuncle) for this project is installed. Please make and install a development build on the device first.
> > > > Learn more: https://docs.expo.dev/development/build/
> > 
> > 
> > this is an issue with the development build not being installed in the simulator
> 
> @Snehal018 , If that is so, why is it working in one PR and not in another? Don't you think the problem might be in the way we mention package dependencies in git commit?

@bharathkumarchowdary, technically if you can run the 45 PR, then you should be able to run this PR as well, can you write steps on how are you running the app?

**Review by bharathkumarchowdary — DISMISSED** [review]

**Review by bharathkumarchowdary — APPROVED** [review]

**Review by mohitkhatri06 — APPROVED** [review]

**Pull request closed without merging**